frontend/app/forms/motion-to-reduce-conviction/motion-to-reduce-conviction.web.component.js

```diff
@@ -1,5 +1,83 @@
+import Radio from "../inputs/radio.component.js";
+import GroupSelect from "../inputs/group-select.component.js";
+import {
+  petitionerRepresentativeOptions,
+  courtTypeOptions,
+  JusticeCourtList,
+  DistrictCourtList
+} from "../form-common-options/form-common-options";
 
-export default function MotionToReduceConviction_Web() {
-  return <div>Web component</div>;
+export default function MotionToReduceConviction_Web({ data }) {
+  return (
+    <FormThatPrints>
+      <Section name="1. Personal Information">
+        <TextInput dataKey="person.firstName" label="First Name" />
+        <TextInput dataKey="person.middleName" label={__("middle name")} />
+        <TextInput dataKey="person.lastName" label="Last Name" />
+        <TextInput dataKey="person.addressStreet" label={__("street")} />
+        <TextInput dataKey="person.addressCity" label={__("city")} />
+        <TextInput dataKey="person.addressState" label={__("state")} />
+        <TextInput dataKey="person.addressZip" label={__("zip")} />
+        <TextInput dataKey="person.homePhone" label={__("home phone")} />
+        <TextInput dataKey="person.email" label={__("email address")} />
+      </Section>
+      <Section name="2. Case Information">
+        <Select
+          dataKey="person.petitionerRepresentative"
+          label={__("are you filling this out for yourself")}
+          options={petitionerRepresentativeOptions}
+        />
+      </Section>
+      {data.person.petitionerRepresentative === "attorney" && (
+        <TextInput
+          dataKey="person.petitionerBarNumber"
+          label="What is your Utah Bar number?"
+        />
+      )}
+      <Radio
+        label="Court Type"
+        dataKey="case.courtType"
+        options={courtTypeOptions}
+      />
+      {data.case.courtType === "District" && (
+        <GroupSelect
+          dataKey="case.courtAddress"
+          label="District Courts"
+          groupOptions={DistrictCourtList}
+        />
+      )}
+      {data.case.courtType === "Justice" && (
+        <GroupSelect
+          dataKey="case.courtAddress"
+          label="Justice Courts"
+          groupOptions={JusticeCourtList}
+        />
+      )}
+      <TextInput dataKey="case.caseNumber" label={__("case number")} />
+      <TextInput dataKey="case.judgeName" label={__("judge full name")} />
+      <TextInput dataKey="case.plaintiff" label={__("plaintiff")} />
+      <TextInput dataKey="case.defendant" label={__("defendant")} />
+      <Select
+        dataKey="case.convictionDegree"
+        label={__("degree of your criminal conviction")}
+        options={convictionDegreeOptions}
+      />
+      <Select
+        dataKey="case.reducedConvictionDegree"
+        label={__("requesting court to reduce my convition to")}
+        options={convictionDegreeOptions}
+      />
+    </FormThatPrints>
+  );
 }
+
+const convictionDegreeOptions = [
+  { label: "First degree", value: "First Degree" },
```

Comment:
Yes, I agree that it would simply the code if we used an int value rather a string. I have no problem with it as long as it doesn't effect process-ability of the form.  I imagine it wouldn't.

Reply:
@scotttball After discovering 4 additional conviction classifications I re-introduced `degreeStringToInt` func. Initially I was only handling 1st, 2nd, and 3rd degree felonies, and it made sense that `degreeStringToInt` was not needed in that case. Now, with the additional misdemeanor classes A, B, C, and Infraction, `degreeStringToInt` function helps calculate difference of these additional convictions to another conviction.